### PR TITLE
charts/yeti Yeti Helm Chart

### LINKED ISSUE #27: Yeti Integration
### Name and Version

charts/osdfir-infrastructure

### What is the problem this feature will solve?

Add [Yeti](https://github.com/yeti-platform/yeti) integration into OSDFIR Infrastructure for open source threat intel monitoring 

FYI @tomchop 

### What is the feature you are proposing to solve the problem?

Specifically create a standalone Yeti Helm chart then integration it as a dependency that can be enabled through OSDFIR Infrastructure or used standalone.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by wajihyassine**
### Applicable issues

- fixes #27 

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using[readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [charts/<name_of_the_chart>] Descriptive title

### CODE REVIEW

File: charts/yeti/templates/api-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        - name: api
+          securityContext:
+              {{- toYaml .Values.api.securityContext | nindent 12 }}
+          image: "{{ .Values.api.image.repository }}:{{ .Values.api.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.api.image.pullPolicy }}
+          command: ["sh", "-c", "/docker-entrypoint.sh webserver"]
+          lifecycle:
+            postStart:
+              exec:
+                command: ["sh", "-c", "poetry run python yetictl/cli.py create-user yeti $YETI_USER --api_key $YETI_API_KEY --admin"]
```
Comment (tomchop) [inline charts/yeti/templates/api-deployment.yaml:41]: So here you are creating a yeti user with username=yeti and password=$YETI_USER which I don't think is what you want. maybe `YETI_USER_PASSWORD`?

**Review by tomchop — CHANGES_REQUESTED** [review]

File: charts/yeti/templates/api-deployment.yaml (hunk 1 above)
Comment (wajihyassine) [reply charts/yeti/templates/api-deployment.yaml]: Ack updated to YETI_USER_PASSWORD so that its more clear thats a password

**Comment by wajihyassine** [comment]
@tomchop as part of your review if you can PTAL at the following:
  *  Upstream fix for the API key generation
  *  Install the chart and just make sure it works properly for you/deployment naming conventions make sense
  *  Glance at the rest of the Yeti Helm Chart values and double check the variable names making sense / necessary config options are exposed

**Review by wajihyassine — COMMENTED** [review]
Fixed

File: charts/osdfir-infrastructure/Chart.yaml [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
-- condition: turbinia.enabled
+  version: 0.3.5
+- condition: global.turbinia.enabled
   name: turbinia
   repository: https://google.github.io/osdfir-infrastructure/
   version: 0.3.3
+- condition: global.yeti.enabled
+  name: yeti
+  repository: https://google.github.io/osdfir-infrastructure/
+  version: 0.1.0
```
Comment (tomchop) [inline charts/osdfir-infrastructure/Chart.yaml:26]: This is the version number of the chart, not Yeti, right?

File: charts/yeti/templates/nginx-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        root /www;
+
+        location /api/v2 {
+            proxy_pass http://{{ include "yeti.fullname" . }}-api:{{ include "yeti.service.port" . }};
+        }
+
+        location / {
+            try_files $uri $uri/ /index.html;
+        }
+    }
```
Comment (tomchop) [inline charts/yeti/templates/nginx-configmap.yaml:21]: There's been a small update to this file! https://github.com/yeti-platform/yeti-docker/blob/main/prod/frontend/nginx.conf

```suggestion
data:
  default.conf: |
    server {

        root /www;

        location /api/v2 {
            proxy_pass http://{{ include "yeti.fullname" . }}-api:{{ include "yeti.service.port" . }};
        }
        
        location ~(^/docs|^/openapi.json) {
            proxy_pass http://api:8000;
        }

        location / {
            try_files $uri $uri/ /index.html;
        }
    }
```

**Review by tomchop — APPROVED** [review]
one nit and one question, otherwise this LGTM!

File: charts/osdfir-infrastructure/Chart.yaml (hunk 2 above)
Comment (wajihyassine) [reply charts/osdfir-infrastructure/Chart.yaml:26]: Yep exactly, although let me bump that to 1.0.0 (going to do that with the rest of the charts as well) given that we have adequate testing in place now

**Review by wajihyassine — COMMENTED** [review]
Awesome thanks for taking a look!

File: charts/osdfir-infrastructure/Chart.yaml (hunk 2 above)
Comment (wajihyassine) [reply charts/osdfir-infrastructure/Chart.yaml:26]: Actually going to do in separate PR along with the other charts

**Pull request merged**